### PR TITLE
Add 2024 changelog

### PR DESCRIPTION
**Pull request by daltzctr**
Still missing API doc links

### CODE REVIEW

File: source/docs/yearly-changes/yearly-changelog.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+.. important:: Both ``DifferentialMechanism`` (Pro only) and ``SimpleDifferentialMechanism`` require `CANFD <https://store.ctr-electronics.com/canivore/>`__.
+
+``DifferentialMechanism`` (Java, C++, Python) provides an easy way to control two-axis differential mechanisms such as two motor elevators (motor on the left and right side of the elevator but is not mechanically linked). Information on this is available here.
+
+New Configs
+^^^^^^^^^^^
+
+We've added several new configs. A full list of available configs is available in the ``configs`` (Java, C++, Python) namespace.
+
```
Comment (bhall-ctre) [inline source/docs/yearly-changes/yearly-changelog.rst:93]: Other major changes we may want to mention:
- New helper methods for working with multiple devices/StatusSignals
- Improvements to status signal update frequency behavior and getting applied update frequency
- Disabling status signals and optimizeBusUtilization
- Orchestra added, now with support for multiple devices playing a single track
- More feedforwards
- Support for roboRIO motion profiles using Velocity/Acceleration setpoints in Position/Velocity controls
- Improved support for unit tests (I already have a PR open to add a page to docs)
- Support for remote limits and more remote sensors
- New Motion Magic control requests

Comment (daltzctr) [reply source/docs/yearly-changes/yearly-changelog.rst:93]: I explicitly didn't include orchestra as that's a port from v5 feature.

File: source/docs/yearly-changes/yearly-changelog.rst [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+* Orchestra have been re-added.
+
+  * Now supports multiple devices playing a single track.
+
+* Remote limits have been re-added.
+* More feedforwards in various controls.
+* Support for roboRIO motion profiles using Velocity/Acceleration setpoints in Position/Velocity controls.
+* Improved support for unit tests.
+* New helper methods for working with multiple signals (single or multi device).
```
Comment (CoryNessCTR) [inline source/docs/yearly-changes/yearly-changelog.rst:127]:
```suggestion
* New helper methods when working with multiple signals (single or multi device).
```

**Review by CoryNessCTR — CHANGES_REQUESTED** [review]

File: source/docs/yearly-changes/yearly-changelog.rst [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For a full list of new motion magic controls, see the controls API documentation (Java, C++, Python).
+
+Differential Mechanisms
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. important:: ``DifferentialMechanism`` requires both Pro and CANFD. ``SimpleDifferentialMechanism`` is a lower performance alternative that requires neither.
+
+``DifferentialMechanism`` (Java, C++, Python) provides an easy way to control two-axis differential mechanisms such as two motor elevators (motor on the left and right side of the elevator but is not mechanically linked). Information on this is available here.
```
Comment (CoryNessCTR) [inline source/docs/yearly-changes/yearly-changelog.rst:109]:
```suggestion
``DifferentialMechanism`` provides an easy way to control two-axis differential mechanisms such as two motor elevators (motor on the left and right side of the elevator but is not mechanically linked). Information on this is available here.
```
We don't have python support yet, so I'd remove the language list (and we kinda have c# support already, both will be fleshed out later, so it's not really important to give a language list)

**Review by CoryNessCTR — APPROVED** [review]

**Review by CoryNessCTR — APPROVED** [review]

File: source/docs/yearly-changes/yearly-changelog.rst [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+Information on configuring the logger and extracting the logs are available in the API Reference and Tuner sections.
+
+.. image:: images/tuner-x-log-extractor.png
+   :alt: Log extractor page in Tuner X
+
+Signal API Improvements
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Users can now disable signals by setting their update rate to 0Hz. To reduce boilerplate when disabling signals, we have added an ``optimizeBusUtilization`` function that takes a list of signals as a parameter. This would automatically disable all signals that are not used.
```
Comment (bhall-ctre) [inline source/docs/yearly-changes/yearly-changelog.rst:85]: optimizeBusUtilization is actually a function on the device, and we have optimizeBusUtilizationForAll which takes a list of devices to optimize.

File: source/docs/yearly-changes/yearly-changelog.rst [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+Additionally, the following new functions have been added.
+
+* ``refreshAll``
+
+  * refreshes all passed in signals
+
+* ``getAppliedUpdateFrequency``
+
+  * retrieves the actual update frequency of a given signal
```
Comment (bhall-ctre) [inline source/docs/yearly-changes/yearly-changelog.rst:97]: One more we can mention is setUpdateFrequencyForAll

**Comment by bhall-ctre** [comment]
Also did we plan for this page to be linked in the sidebar, or just from the changelog link on the home page? Doesn't look like we added it to the toc.

**Pull request merged**